### PR TITLE
Bug/storage sync int overflow

### LINKED ISSUE #28564: UserStorageSyncManager int overflow
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

storage

### Describe the bug

Keycloak crashes when sync periods are set to 2147484 or greater due to an integer overflow

### Version

24.0.2

### Regression

- [ ] The issue is a regression

### Expected behavior

I should be able to set sync periods of greater than 2147484 without Keycloak crashing.

### Actual behavior

When sync periods are set to 2147484 or greater, an integer overflow occurs due to the multiplication, resulting in a negative delay being passed to the timer despite the checks lines [207](https://github.com/tqe1999/keycloak/blob/1bb5e1413416cbe7a83952c6a01e7f75ac7502f3/model/storage-private/src/main/java/org/keycloak/storage/managers/UserStorageSyncManager.java#L207) and [217](https://github.com/tqe1999/keycloak/blob/1bb5e1413416cbe7a83952c6a01e7f75ac7502f3/model/storage-private/src/main/java/org/keycloak/storage/managers/UserStorageSyncManager.java#L217). As a result Keycloak crashes due to the unhandled exception.

### How to Reproduce?

Set a sync period to 2147484 or greater, then restart Keycloak.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by tqe1999**
Fixes integer overflow with explicit cast to long.

Bug reproduced by failing test at https://github.com/keycloak/keycloak/commit/7cc8a320778bfbb3b33157706a2253a230b4914a

Closes https://github.com/keycloak/keycloak/issues/28564

### CODE REVIEW

**Review by ahus1 — APPROVED** [review]
Thank you for fixing this!

**Comment by ahus1** [comment]
@martin-kanis - Oh, just saw that you self-assigned it, I didn't notice it as I followed a link from my inbox. I will wait for your review before merging.

**Review by martin-kanis — APPROVED** [review]
@tqe1999 Thanks for the issue and PR. Looks good.
@ahus1 you can proceed :)

**Comment by PerilousApricot** [comment]
Hi @tqe1999 I've hit this error and came here from StackOverflow. Do you happen to know what table holds these sync values? We're unable to use the fix until it's released and added to a bitnami helm release, so I'm hoping that some database surgery will let us get our instance restarted.

**Comment by mhajas** [comment]
Hello @PerilousApricot, the value is stored in the table `component_config`. You are looking for rows with `name` equal to `fullSyncPeriod` or `changedSyncPeriod`. If it works for you, please comment on the issue so people can use it as a workaround.

**Pull request merged**
